src/main/scala/rocketchip/RISCVPlatform.scala

```diff
@@ -4,153 +4,126 @@ package rocketchip
 
 import Chisel._
 import config._
+import coreplex._
 import diplomacy._
+import jtag.JTAGIO
 import uncore.tilelink2._
 import uncore.devices._
 import util._
-import jtag.JTAGIO
-import coreplex._
 
-// System with JTAG DTM Instantiated inside. JTAG interface is
-// exported outside.
-
-trait PeripheryJTAGDTM extends HasTopLevelNetworks {
-  val module: PeripheryJTAGDTMModule
+/** All the traits defined in this file assume that they are being mixed in
+  * to a system that has a standard RISCV-based coreplex platform.
+  */
+trait HasCoreplexRISCVPlatform {
   val coreplex: CoreplexRISCVPlatform
 }
 
-trait PeripheryJTAGDTMBundle extends HasTopLevelNetworksBundle {
-  val outer: PeripheryJTAGDTM
-
-  val jtag = new JTAGIO(hasTRSTn = false).flip
-  val jtag_reset = Bool(INPUT)
-  val jtag_mfr_id = UInt(INPUT, 11)
-
+/** Adds JTAG DTM to system, and exports JTAG interface. */
+trait HasPeripheryJTAGDTM extends HasSystemNetworks with HasCoreplexRISCVPlatform {
+  val module: HasPeripheryJTAGDTMModuleImp
 }
 
-trait PeripheryJTAGDTMModule extends HasTopLevelNetworksModule {
-  val outer: PeripheryJTAGDTM
-  val io: PeripheryJTAGDTMBundle
+trait HasPeripheryJTAGDTMModuleImp extends LazyMultiIOModuleImp {
+  val outer: HasPeripheryJTAGDTM
+
+  val jtag = IO(new JTAGIO(hasTRSTn = false).flip)
```

Comment:
BTW, seems like some of the code below related to the optional inclusion could be simplified if there was a top level bundle called `io_jtag` with three members: `reset: Bool`, `mfr_id: UInt` and `<something>: JTAGIO`. Any thoughts there? (I assumed there's a bunch of wrapper code expecting `io_jtag_TCK` &etc so I kept the hierarchy the same, though I did mess up the names...)